addon/components/abstract-modal.js

```diff
@@ -0,0 +1,139 @@
+/* global Ps */
+
+/**
+ * Functionality to style headder/footer elements depending on scroll state, content hidden under header/footer
+ */
+
+import Ember from 'ember'
+import $ from 'jquery'
+
+const {Component} = Ember
+
+export default Component.extend({
+  scrollBindingsSet: false,
+  containerObserver: null,  // @type {window.MutationObserver}
+  $containerEl: null, // @type {JQuery object}
+  $headerEl: null, // @type {JQuery object}
+  $footerEl: null, // @type {JQuery object}
+  containerElClassname: 'ps-container',
+  headerElClassname: 'input-header',
+  footerElClassname: 'actions',
+
+  /**
+   * create a window.MutationObserver on the container element
+   * mutation events callback invokes updateScrollStyles()
+   * @returns {MutationObserver} window.MutationObserver on container element
+   */
+
+  createMutationObserver () {
+    // create an observer instance
+    let mutationObserver = new window.MutationObserver((mutations) => {
+      mutations.forEach((mutation) => {
+        this.updateScrollStyles()
+      })
+    })
+
+    // configuration of the observer
+    const config = { attributes: false, childList: true, characterData: false, subtree: true }
+
+    // pass in the target DOM node, as well as the observer options
+    mutationObserver.observe(this.get('$containerEl').get(0), config)
+
+    return mutationObserver
+  },
+
+  /**
+  * update perfect-scrollbar before checking scoll position
+  * need to disconnect MutationObserver then re-establish to prevent infinite loop
+  * of mutation event triggers, callback updates, which triggers new mutation event...
+  */
+  updatePerfectScrollbar () {
+    let mutationObserver
+    const $containerEl = this.get('$containerEl')
+
+    Ps.update($containerEl.get(0))
+
+    if (this.get('containerObserver')) {
+      this.get('containerObserver').disconnect()
+      mutationObserver = this.createMutationObserver()
+    }
+
+    if (this.isDestroyed || this.isDestroying) {
```

Comment:
@sandersky:  I think you raise a good point.  In this conditional, before the return I should have:
mutationObserver.disconnect()

otherwise, since I have in ln 65:
this.set('containerObserver', mutationObserver)

won't the disconnect be taken care of in ln 120?
if (this.get('containerObserver')) {
  this.get('containerObserver').disconnect()
 }